### PR TITLE
WFLY-18091 Upgrade Infinispan to 14.0.10.Final

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-18091

N.B. This PR fixes indentation in the licences.xml for ee-feature-pack, as discussed here:
https://wildfly.zulipchat.com/#narrow/stream/174184-wildfly-developers/topic/Formatting.20of.20feature.20pack.20licencses.2Exml

### CODE REVIEW

**Review by rhusar — CHANGES_REQUESTED** [review]
Causes failures in org.jboss.as.test.clustering.cluster.jms.ClusteredMessagingTestCase#
testClusteredQueue and #testClusteredTopic

**Comment by darranl** [comment]
Hi @rhusar is this one possible to get a review?

**Review by rhusar — APPROVED** [review]
LGTM

File: testsuite/integration/clustering/pom.xml [hunk 1]
```diff
@@ -257,6 +257,10 @@
             <artifactId>infinispan-client-hotrod-jakarta</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.infinispan</groupId>
+            <artifactId>infinispan-clustered-counter</artifactId>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (pferraro) [reply testsuite/integration/clustering/pom.xml:288]: Indeed. I'll fix this later.

**Pull request merged**